src/main/resources/resource/WebGui/service/js/ProgramABGui.js

```diff
@@ -29,6 +29,10 @@ angular.module('mrlapp.service.ProgramABGui', [])
         // all service data should never be written to, only read from
         $scope.currentUserName = service.currentUserName;
         $scope.currentBotName = service.currentBotName;
+        $scope.conversation = service.conversationHistory;
+                  for (var i = 0; i < $scope.conversation.length; i++) {
+        $scope.conversation[i] = $sce.trustAsHtml($scope.conversation[i]);

```

Comment:
Uh, this will kill your performance, but as long as there are no other services doing the same, it should be fine.
You're re-trusting all messages everytime a new one arrives.

Reply:
Agreed ...  the UI should just be initializing the conversation list on inital load.. then subscribing to every time a new response is published. and appending that specific response.